File: complex-events-backend/app/data/mock_data.py
MOCK_EQUIPMENT_CATEGORIES = [
    {"value": "动设备", "label": "动设备", "count": 5},
    {"value": "静设备", "label": "静设备", "count": 4},
    {"value": "电气设备", "label": "电气设备", "count": 3},
    {"value": "管道设备", "label": "管道设备", "count": 2},
]

MOCK_EQUIPMENT_TYPES = {
    "动设备": ["离心泵", "压缩机", "风机", "齿轮泵", "真空泵"],
    "静设备": ["储罐", "换热器", "反应器", "塔器"],
    "电气设备": ["变压器", "配电柜", "电动机"],
    "管道设备": ["阀门", "管道"]
}

MOCK_EQUIPMENT_INSTANCES = {
    "动设备|离心泵": [
        {"id": 1, "name": "P01 离心泵"},
        {"id": 2, "name": "P02 离心泵"},
    ],
    "动设备|压缩机": [
        {"id": 3, "name": "C01 压缩机"},
    ],
    "动设备|风机": [
        {"id": 4, "name": "F01 风机"},
    ],
    "动设备|齿轮泵": [
        {"id": 5, "name": "G01 齿轮泵"},
    ],
    "静设备|储罐": [
        {"id": 6, "name": "TK01 储罐"},
        {"id": 7, "name": "TK02 储罐"},
    ],
    "静设备|换热器": [
        {"id": 8, "name": "E01 换热器"},
    ],
    "静设备|反应器": [
        {"id": 9, "name": "R01 反应器"},
    ],
    "电气设备|变压器": [
        {"id": 10, "name": "T01 变压器"},
    ],
    "电气设备|配电柜": [
        {"id": 11, "name": "PC01 配电柜"},
        {"id": 12, "name": "PC02 配电柜"},
    ],
    "管道设备|阀门": [
        {"id": 13, "name": "V01 阀门"},
        {"id": 14, "name": "V02 阀门"},
    ],
}

MOCK_PROCESSES = [
    {"id": 1, "equipment_id": 1, "equipment_name": "P01 离心泵", "equipment_category": "动设备", "equipment_type_name": "离心泵", "process_name": "离心泵检修", "status": "on_hold", "estimated_hours": 8, "scheduled_start_time": "2026-03-12 08:00:00", "scheduled_end_time": "2026-03-12 16:00:00", "description": "定期检修离心泵，检查轴承和密封", "is_milestone": True},
    {"id": 2, "equipment_id": 2, "equipment_name": "P02 离心泵", "equipment_category": "动设备", "equipment_type_name": "离心泵", "process_name": "离心泵保养", "status": "pending", "estimated_hours": 4, "scheduled_start_time": "2026-03-13 08:00:00", "scheduled_end_time": "2026-03-13 12:00:00", "description": "日常保养，更换润滑油", "is_milestone": False},
    {"id": 3, "equipment_id": 3, "equipment_name": "C01 压缩机", "equipment_category": "动设备", "equipment_type_name": "压缩机", "process_name": "压缩机大修", "status": "completed", "estimated_hours": 24, "scheduled_start_time": "2026-03-10 08:00:00", "scheduled_end_time": "2026-03-11 08:00:00", "description": "全面大修，更换转子和轴承", "is_milestone": True},
    {"id": 4, "equipment_id": 6, "equipment_name": "TK01 储罐", "equipment_category": "静设备", "equipment_type_name": "储罐", "process_name": "储罐清洗", "status": "on_hold", "estimated_hours": 12, "scheduled_start_time": "2026-03-14 08:00:00", "scheduled_end_time": "2026-03-14 20:00:00", "description": "内部清洗和防腐处理", "is_milestone": False},
    {"id": 5, "equipment_id": 8, "equipment_name": "E01 换热器", "equipment_category": "静设备", "equipment_type_name": "换热器", "process_name": "换热器检修", "status": "pending", "estimated_hours": 6, "scheduled_start_time": "2026-03-15 08:00:00", "scheduled_end_time": "2026-03-15 14:00:00", "description": "检查换热管，清理污垢", "is_milestone": False},
    {"id": 6, "equipment_id": 9, "equipment_name": "R01 反应器", "equipment_category": "静设备", "equipment_type_name": "反应器", "process_name": "反应器维护", "status": "in_progress", "estimated_hours": 16, "scheduled_start_time": "2026-03-11 08:00:00", "scheduled_end_time": "2026-03-12 00:00:00", "description": "催化剂更换和设备检查", "is_milestone": True},
    {"id": 7, "equipment_id": 10, "equipment_name": "T01 变压器", "equipment_category": "电气设备", "equipment_type_name": "变压器", "process_name": "变压器检测", "status": "on_hold", "estimated_hours": 4, "scheduled_start_time": "2026-03-16 08:00:00", "scheduled_end_time": "2026-03-16 12:00:00", "description": "绝缘测试和油质分析", "is_milestone": False},
    {"id": 8, "equipment_id": 11, "equipment_name": "PC01 配电柜", "equipment_category": "电气设备", "equipment_type_name": "配电柜", "process_name": "配电柜检修", "status": "rejected", "estimated_hours": 2, "scheduled_start_time": "2026-03-17 08:00:00", "scheduled_end_time": "2026-03-17 10:00:00", "description": "检查接线和保护装置", "is_milestone": False, "approval_comments": "需要补充材料"},
    {"id": 9, "equipment_id": 13, "equipment_name": "V01 阀门", "equipment_category": "管道设备", "equipment_type_name": "阀门", "process_name": "阀门更换", "status": "pending", "estimated_hours": 3, "scheduled_start_time": "2026-03-18 08:00:00", "scheduled_end_time": "2026-03-18 11:00:00", "description": "更换损坏的阀门", "is_milestone": False},
    {"id": 10, "equipment_id": 4, "equipment_name": "F01 风机", "equipment_category": "动设备", "equipment_type_name": "风机", "process_name": "风机平衡", "status": "in_progress", "estimated_hours": 6, "scheduled_start_time": "2026-03-12 08:00:00", "scheduled_end_time": "2026-03-12 14:00:00", "description": "动平衡测试和校正", "is_milestone": False},
    {"id": 11, "equipment_id": 5, "equipment_name": "G01 齿轮泵", "equipment_category": "动设备", "equipment_type_name": "齿轮泵", "process_name": "齿轮泵维修", "status": "on_hold", "estimated_hours": 5, "scheduled_start_time": "2026-03-19 08:00:00", "scheduled_end_time": "2026-03-19 13:00:00", "description": "修复齿轮磨损问题", "is_milestone": False},
    {"id": 12, "equipment_id": 7, "equipment_name": "TK02 储罐", "equipment_category": "静设备", "equipment_type_name": "储罐", "process_name": "储罐检测", "status": "completed", "estimated_hours": 8, "scheduled_start_time": "2026-03-08 08:00:00", "scheduled_end_time": "2026-03-08 16:00:00", "description": "壁厚检测和安全评估", "is_milestone": False},
]

MOCK_WORKERS = [
    {"id": 1, "name": "张三", "role": "高级工程师", "status": "工作中", "organization": "维修部", "skill_level": "高级", "tasks": [{"task_id": 1, "task_name": "离心泵检修", "equipment": "P01 离心泵", "start_time": "第1天 08:00", "end_time": "第1天 16:00", "status": "进行中"}, {"task_id": 6, "task_name": "反应器维护", "equipment": "R01 反应器", "start_time": "第1天 08:00", "end_time": "第2天 00:00", "status": "进行中"}]},
    {"id": 2, "name": "李四", "role": "技术员", "status": "工作中", "organization": "维修部", "skill_level": "中级", "tasks": [{"task_id": 2, "task_name": "离心泵保养", "equipment": "P02 离心泵", "start_time": "第2天 08:00", "end_time": "第2天 12:00", "status": "进行中"}]},
    {"id": 3, "name": "王五", "role": "初级工程师", "status": "空闲中", "organization": "电气部", "skill_level": "初级", "tasks": []},
    {"id": 4, "name": "赵六", "role": "高级工程师", "status": "工作中", "organization": "工艺部", "skill_level": "高级", "tasks": [{"task_id": 4, "task_name": "储罐清洗", "equipment": "TK01 储罐", "start_time": "第3天 08:00", "end_time": "第3天 20:00", "status": "进行中"}]},
    {"id": 5, "name": "孙七", "role": "技术员", "status": "空闲中", "organization": "管道部", "skill_level": "中级", "tasks": []},
    {"id": 6, "name": "周八", "role": "设备管理员", "status": "工作中", "organization": "设备部", "skill_level": "中级", "tasks": [{"task_id": 7, "task_name": "变压器检测", "equipment": "T01 变压器", "start_time": "第5天 08:00", "end_time": "第5天 12:00", "status": "进行中"}]},
]

MOCK_ORDERS = [
    {"work_order_id": "WO001", "order_number": "WO-2024-001", "process_name": "离心泵检修", "status": "on_hold", "work_order_status": "on_hold", "equipment_id": "E001", "equipment_name": "P01 离心泵", "priority": "高"},
    {"work_order_id": "WO002", "order_number": "WO-2024-002", "process_name": "压缩机大修", "status": "completed", "work_order_status": "completed", "equipment_id": "E003", "equipment_name": "C01 压缩机", "priority": "高"},
    {"work_order_id": "WO003", "order_number": "WO-2024-003", "process_name": "储罐清洗", "status": "on_hold", "work_order_status": "on_hold", "equipment_id": "E006", "equipment_name": "TK01 储罐", "priority": "中"},
    {"work_order_id": "WO004", "order_number": "WO-2024-004", "process_name": "变压器检测", "status": "pending", "work_order_status": "pending", "equipment_id": "E010", "equipment_name": "T01 变压器", "priority": "低"},
]

MOCK_MATERIALS = [
    {"material_name": "润滑油", "initial_stock": 100.0, "current_stock": 70.0, "unit": "升"},
    {"material_name": "密封件", "initial_stock": 50.0, "current_stock": 35.0, "unit": "个"},
    {"material_name": "轴承", "initial_stock": 30.0, "current_stock": 20.0, "unit": "套"},
    {"material_name": "垫片", "initial_stock": 200.0, "current_stock": 150.0, "unit": "片"},
    {"material_name": "螺栓", "initial_stock": 500.0, "current_stock": 400.0, "unit": "个"},
]

MOCK_TOOLS = [
    {"tool_id": "TL001", "tool_name": "扳手组", "tool_type": "手动工具", "usage_status": "占用", "usage_tasks": [{"task_name": "离心泵检修"}]},
    {"tool_id": "TL002", "tool_name": "万用表", "tool_type": "测量工具", "usage_status": "空闲", "usage_tasks": []},
    {"tool_id": "TL003", "tool_name": "起重机", "tool_type": "起重设备", "usage_status": "占用", "usage_tasks": [{"task_name": "压缩机大修"}]},
    {"tool_id": "TL004", "tool_name": "液压扳手", "tool_type": "动力工具", "usage_status": "空闲", "usage_tasks": []},
    {"tool_id": "TL005", "tool_name": "探伤仪", "tool_type": "检测设备", "usage_status": "空闲", "usage_tasks": []},
]

def get_mock_equipment_info():
    return {
        "categories": MOCK_EQUIPMENT_CATEGORIES,
        "types": MOCK_EQUIPMENT_TYPES,
        "instances": MOCK_EQUIPMENT_INSTANCES
    }

def get_mock_process_list(filters=None):
    processes = MOCK_PROCESSES
    if filters:
        if filters.get('equipment_category'):
            processes = [p for p in processes if p['equipment_category'] == filters['equipment_category']]
        if filters.get('equipment_type'):
            processes = [p for p in processes if p['equipment_type_name'] == filters['equipment_type']]
        if filters.get('equipment_id'):
            processes = [p for p in processes if p['equipment_id'] == int(filters['equipment_id'])]
        if filters.get('status'):
            processes = [p for p in processes if p['status'] == filters['status']]
    return {"total": len(processes), "list": processes}

def get_mock_process_detail(process_id):
    for p in MOCK_PROCESSES:
        if p['id'] == process_id:
            return p
    return None

def get_mock_workers():
    return {"success": True, "worker_status": MOCK_WORKERS}

def get_mock_orders():
    return {"success": True, "data": MOCK_ORDERS}

def get_mock_materials():
    return {"success": True, "material_inventory": MOCK_MATERIALS}

def get_mock_tools():
    return {"success": True, "maintenance_tool_status": MOCK_TOOLS}

File: complex-events-frontend/src/views/InfoPanel/data/mockData.js
export const mockWorkers = [
  {
    worker_id: 'W001',
    worker_name: '张三',
    worker_type: '钳工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T001',
        task_name: '空冷器1检修',
        status: '进行中',
        equipment: '空冷器1',
        start_time: '第1天 08:00',
        end_time: '第1天 12:00'
      },
      {
        task_id: 'T002',
        task_name: '离心泵1检修',
        status: '未开始',
        equipment: '离心泵1',
        start_time: '第2天 08:00',
        end_time: '第2天 16:00'
      }
    ]
  },
  {
    worker_id: 'W002',
    worker_name: '李四',
    worker_type: '焊工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T003',
        task_name: '轴流式通风机1检修',
        status: '进行中',
        equipment: '轴流式通风机1',
        start_time: '第1天 09:00',
        end_time: '第1天 17:00'
      }
    ]
  },
  {
    worker_id: 'W003',
    worker_name: '王五',
    worker_type: '起重工',
    status: '空闲中',
    tasks: []
  },
  {
    worker_id: 'W004',
    worker_name: '赵六',
    worker_type: '电工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T004',
        task_name: '空冷器1检修',
        status: '进行中',
        equipment: '空冷器1',
        start_time: '第1天 10:00',
        end_time: '第1天 15:00'
      },
      {
        task_id: 'T005',
        task_name: '离心泵1检修',
        status: '未开始',
        equipment: '离心泵1',
        start_time: '第3天 08:00',
        end_time: '第3天 12:00'
      }
    ]
  },
  {
    worker_id: 'W005',
    worker_name: '钱七',
    worker_type: '管工',
    status: '已完成',
    tasks: [
      {
        task_id: 'T006',
        task_name: '空冷器1检修',
        status: '已完成',
        equipment: '空冷器1',
        start_time: '第1天 08:00',
        end_time: '第1天 11:00'
      }
    ]
  },
  {
    worker_id: 'W006',
    worker_name: '孙八',
    worker_type: '仪表工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T007',
        task_name: '轴流式通风机1检修',
        status: '进行中',
        equipment: '轴流式通风机1',
        start_time: '第1天 08:00',
        end_time: '第1天 16:00'
      }
    ]
  },
  {
    worker_id: 'W007',
    worker_name: '周九',
    worker_type: '车工',
    status: '空闲中',
    tasks: []
  },
  {
    worker_id: 'W008',
    worker_name: '吴十',
    worker_type: '铣工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T008',
        task_name: '离心泵1检修',
        status: '进行中',
        equipment: '离心泵1',
        start_time: '第1天 13:00',
        end_time: '第1天 18:00'
      }
    ]
  },
  {
    worker_id: 'W009',
    worker_name: '郑一',
    worker_type: '钻工',
    status: '空闲中',
    tasks: []
  },
  {
    worker_id: 'W010',
    worker_name: '王二',
    worker_type: '刨工',
    status: '工作中',
    tasks: [
      {
        task_id: 'T009',
        task_name: '空冷器1检修',
        status: '进行中',
        equipment: '空冷器1',
        start_time: '第1天 14:00',
        end_time: '第1天 20:00'
      }
    ]
  }
]

export const mockOrders = [
  {
    work_order_id: 'WO001',
    order_number: 'WO-2024-001',
    process_name: '空冷器1检修',
    status: 'construction_confirmed',
    work_order_status: 'construction_confirmed',
    equipment_id: 'E001',
    equipment_name: '空冷器1',
    priority: '高',
    workers: {
      '钳工': ['张三'],
      '电工': ['赵六'],
      '管工': ['钱七'],
      '刨工': ['王二']
    }
  },
  {
    work_order_id: 'WO002',
    order_number: 'WO-2024-002',
    process_name: '轴流式通风机1检修',
    status: 'process_closed',
    work_order_status: 'process_closed',
    equipment_id: 'E002',
    equipment_name: '轴流式通风机1',
    priority: '中',
    workers: {
      '焊工': ['李四'],
      '仪表工': ['孙八']
    }
  },
  {
    work_order_id: 'WO003',
    order_number: 'WO-2024-003',
    process_name: '离心泵1检修',
    status: 'eng_approved',
    work_order_status: 'eng_approved',
    equipment_id: 'E003',
    equipment_name: '离心泵1',
    priority: '高',
    workers: {
      '钳工': ['张三'],
      '电工': ['赵六'],
      '铣工': ['吴十']
    }
  },
  {
    work_order_id: 'WO004',
    order_number: 'WO-2024-004',
    process_name: '换热器1清洗',
    status: 'released',
    work_order_status: 'released',
    equipment_id: 'E004',
    equipment_name: '换热器1',
    priority: '低',
    workers: {
      '管工': ['钱七']
    }
  },
  {
    work_order_id: 'WO005',
    order_number: 'WO-2024-005',
    process_name: '压缩机1检修',
    status: 'construction_signed',
    work_order_status: 'construction_signed',
    equipment_id: 'E005',
    equipment_name: '压缩机1',
    priority: '高',
    workers: {
      '钳工': ['张三', '王五'],
      '电工': ['赵六'],
      '焊工': ['李四']
    }
  },
  {
    work_order_id: 'WO006',
    order_number: 'WO-2024-006',
    process_name: '阀门组更换',
    status: 'equipment_closed',
    work_order_status: 'equipment_closed',
    equipment_id: 'E006',
    equipment_name: '阀门组A',
    priority: '中',
    workers: {
      '管工': ['钱七'],
      '钳工': ['周九']
    }
  }
]

export const mockMaterials = [
  {
    material_name: '不锈钢管',
    initial_stock: 100,
    current_stock: 35,
    unit: '米'
  },
  {
    material_name: '法兰盘',
    initial_stock: 50,
    current_stock: 40,
    unit: '个'
  },
  {
    material_name: '螺栓螺母',
    initial_stock: 500,
    current_stock: 200,
    unit: '套'
  },
  {
    material_name: '密封垫片',
    initial_stock: 200,
    current_stock: 50,
    unit: '片'
  },
  {
    material_name: '钢材',
    initial_stock: 2000,
    current_stock: 1500,
    unit: '公斤'
  },
  {
    material_name: '密封胶',
    initial_stock: 50,
    current_stock: 30,
    unit: '支'
  },
  {
    material_name: '电缆',
    initial_stock: 500,
    current_stock: 450,
    unit: '米'
  },
  {
    material_name: '仪表配件',
    initial_stock: 100,
    current_stock: 80,
    unit: '件'
  },
  {
    material_name: '泵体密封件',
    initial_stock: 30,
    current_stock: 10,
    unit: '套'
  },
  {
    material_name: '管道弯头',
    initial_stock: 200,
    current_stock: 180,
    unit: '个'
  }
]

export const mockTools = [
  {
    tool_id: 'TL001',
    tool_name: '电动葫芦',
    tool_type: '起重设备',
    usage_status: '占用',
    usage_tasks: [{ task_name: '空冷器1检修' }]
  },
  {
    tool_id: 'TL002',
    tool_name: '叉车',
    tool_type: '运输设备',
    usage_status: '空闲',
    usage_tasks: []
  },
  {
    tool_id: 'TL003',
    tool_name: '电焊机',
    tool_type: '焊接设备',
    usage_status: '占用',
    usage_tasks: [{ task_name: '轴流式通风机1检修' }]
  },
  {
    tool_id: 'TL004',
    tool_name: '氩弧焊机',
    tool_type: '焊接设备',
    usage_status: '空闲',
    usage_tasks: []
  },
  {
    tool_id: 'TL005',
    tool_name: '通风机',
    tool_type: '通风设备',
    usage_status: '占用',
    usage_tasks: [{ task_name: '空冷器1检修' }]
  },
  {
    tool_id: 'TL006',
    tool_name: '加热器',
    tool_type: '加热设备',
    usage_status: '空闲',
    usage_tasks: []
  },
  {
    tool_id: 'TL007',
    tool_name: '桥式起重机',
    tool_type: '起重设备',
    usage_status: '占用',
    usage_tasks: [{ task_name: '压缩机1检修' }]
  },
  {
    tool_id: 'TL008',
    tool_name: '搬运车',
    tool_type: '运输设备',
    usage_status: '空闲',
    usage_tasks: []
  },
  {
    tool_id: 'TL009',
    tool_name: '切割机',
    tool_type: '焊接设备',
    usage_status: '占用',
    usage_tasks: [{ task_name: '离心泵1检修' }]
  },
  {
    tool_id: 'TL010',
    tool_name: '送风设备',
    tool_type: '通风设备',
    usage_status: '空闲',
    usage_tasks: []
  }
]